lib/ansible/modules/files/synchronize.py

```diff
@@ -44,9 +44,19 @@
   mode:
     description:
       - Specify the direction of the synchronization. In push mode the localhost or delegate is the source; In pull mode the remote host in context is the source.
+      - This option has been deprecated in favor of I(direction)
     required: false
     choices: [ 'push', 'pull' ]
     default: 'push'
+  direction:
```

Comment:
It would conflict with "mode" (i.e. explicitly set Unix permissions), if the synchronize module ever supported it in the same way as other file modules.